File: UI/src/models/RoomDTO.ts
import type { IFilterDTO } from "./base.model";
import type { IBookingDTO } from "./BookingDTO";
import type { IDTO } from "./DTO.model";

export const ROOM_STATUS = ["Active", "Inactive", "Maintenance"] as const;
export type TRoomStatus = (typeof ROOM_STATUS)[number];
export const ROOM_STATUS_LABELS: Record<TRoomStatus, string> = {
  Active: "פעיל",
  Inactive: "לא פעיל",
  Maintenance: "תחזוקה",
};

export interface IRoomDTO extends IDTO {
  name?: string;

  status?: TRoomStatus;
  capacity?: number;
  floor?: number;
  hasProjector?: boolean;
  hasTeamMeeting?: boolean;
  hasConferenceCall?: boolean;
  bookings?: IBookingDTO[] | null;
  imageUrl?: string;
  notes?: string;
}
export interface IRoomEditDTO extends Omit<IRoomDTO, "bookings"> {
  rawImgFile?: File | null;
}

export interface IRoomFilter extends IFilterDTO {
  startTime?: string | null;
  startDate?: string | null | Date;
  endTime?: string | null ;
  endDate?: string | null | Date;
  capacity?: number | null;
  floor?: number | null;
  hasProjector?: boolean | null;
  hasTeamMeeting?: boolean | null;
  hasConferenceCall?: boolean | null;
  status?: TRoomStatus | null;
  roomName?: string | null;
}

File: API/Dtos/Room/RoomDto.cs
using API.Dtos.Booking;
using API.Enums;
using API.Models;

namespace API.Dtos.Room;

public class RoomDto : Dto
{
    public required string Name { get; set; }
    public required int Capacity { get; set; }
    public required int Floor { get; set; }
    public required bool HasProjector { get; set; }
    public required bool HasTeamMeeting { get; set; }
    public required bool HasConferenceCall { get; set; }
    public string? ImageUrl { get; set; }
    public string? PublicCloudinaryId { get; set; }
    public string? Notes { get; set; }
    public required RoomStatus Status { get; set; }
    public List<BookingDto>? Bookings { get; set; }
}